src/network/getters/getSponsorshipDailyBuckets.ts

```diff
@@ -0,0 +1,33 @@
+import getGraphClient from '~/getters/getGraphClient'
+import {
+    GetSponsorshipDailyBucketsDocument,
+    GetSponsorshipDailyBucketsQuery,
+    GetSponsorshipDailyBucketsQueryVariables,
+    SponsorshipDailyBucket_Filter,
+} from '~/generated/gql/network'
+
+export const getSponsorshipDailyBuckets = async (
+    sponsorshipId: string,
+    first: number,
+    skip = 0,
+    dateLowerThan?: string, // utc timestamp in seconds
+): Promise<GetSponsorshipDailyBucketsQuery['sponsorshipDailyBuckets']> => {
+    const where: SponsorshipDailyBucket_Filter = { sponsorship_: { id: sponsorshipId } }
+    if (dateLowerThan) {
+        where.date_lt = dateLowerThan
```

Comment:
Could it be based solely on a range of timestamps? The idea would be to simplify `getSponsorshipStats`.

```ts
async function getSponsorshipDailyBuckets(
    sponsorshipId: string,
    fromTimestamp: number,
    toTimestamp: number
) {
    // use where.date_lt and where.date_gte
}
```

Then it'd be


```ts
const now = Date.now()

// 7d
getSponsorshipDailyBuckets('0x123', beginningOfDay(now - 7d), Date.now())

// ytd
getSponsorshipDailyBuckets('0x123', beginningOfYear(now), Date.now())
```

If you want to skip "today" then you just use `beginningOfDay(now)` at the end.

@pmichalowski thoughts?

Reply:
And range edges could be optional for things like `ChartPeriod.All`.

Reply:
That might make sense in case if for some reason there would be 2 "buckets" per day or if there would be none on some day